### ISSUE TITLE
[API] Improve HTTP header Set/Add API to return error

### ISSUE THREAD

**Issue by chamil321** [ballerina-platform/ballerina-lang#20974]
**Description:**
$Subject as errors are getting paniced and logged in the internal log. Ideally these methods should return an error if operation failed

**Steps to reproduce:**
```
import ballerina/http;

service backend on new http:Listener(9109, {httpVersion: "2.0"}) {
    resource function echoResponseWithTrailer(http:Caller caller, http:Request request) {
        http:Response response = new;
        var textPayload = request.getTextPayload();
        string inPayload = textPayload is string ? textPayload : "error in accessing payload";
        response.setTextPayload(<@untainted> inPayload);
        response.setHeader("Transfer-encoding", "gzip", position = "trailing");
        var result = caller->respond(response);
    }
}
```
Exception is thrown

```
[2020-02-11 12:14:21,878] SEVERE {b7a.log.crash} - prohibited trailing header: Transfer-encoding 
java.lang.IllegalArgumentException: prohibited trailing header: Transfer-encoding
	at io.netty.handler.codec.http.DefaultLastHttpContent$TrailingHttpHeaders$1.validateName(DefaultLastHttpContent.java:125)
	at io.netty.handler.codec.http.DefaultLastHttpContent$TrailingHttpHeaders$1.validateName(DefaultLastHttpContent.java:118)
	at io.netty.handler.codec.DefaultHeaders.set(DefaultHeaders.java:430)
	at io.netty.handler.codec.DefaultHeaders.setObject(DefaultHeaders.java:481)
	at io.netty.handler.codec.http.DefaultHttpHeaders.set(DefaultHttpHeaders.java:165)
	at org.ballerinalang.mime.nativeimpl.EntityHeaders.setHeader(EntityHeaders.java:128)
	at ballerina.mime.natives.externSetHeader(natives.bal:557)
	at ballerina.mime.$value$Entity.setHeader(natives.bal:441)
	at ballerina.mime.$value$Entity.call(natives.bal)
	at ballerina.http.$value$Response.setHeader(http_response.bal:129)
	at ballerina.http.$value$Response.call(http_response.bal)
	at chamil.mytest.$value$backend__service_1.echoResponseWithTrailer(main.bal:48)
	at chamil.mytest.$value$backend__service_1.call(main.bal)
	at org.ballerinalang.jvm.values.connector.Executor.lambda$submit$0(Executor.java:76)
	at org.ballerinalang.jvm.scheduling.SchedulerItem.execute(Scheduler.java:426)
	at org.ballerinalang.jvm.scheduling.Scheduler.run(Scheduler.java:218)
	at org.ballerinalang.jvm.scheduling.Scheduler.runSafely(Scheduler.java:191)
	at java.lang.Thread.run(Thread.java:748)
 

```

**Affected Versions:**
1.1.1

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by chamil321** [comment]
Since 90% of the time addHeader, setHeader APIs do not throw errors, IMO it would be better to catch the error at Java level and panic it properly after creating ballerina error. So it will appear in the console without directing them to the internal-log.

**Comment by chamil321** [comment]
Fixed by #22674, hence closing.

**Comment by chamil321** [comment]
Improve all header related methods to return errors rather than panic
